loopin.go

```diff
@@ -624,8 +624,35 @@ func (s *loopInSwap) executeSwap(globalCtx context.Context) error {
 	// Verify that the confirmed (external) htlc value matches the swap
 	// amount. Otherwise, fail the swap immediately.
 	if htlcValue != s.LoopInContract.AmountRequested {
+		// If the sweep tx can't be published or the context is
+		// cancelled we will set the swap to StateFailIncorrectHtlcAmt.
 		s.setState(loopdb.StateFailIncorrectHtlcAmt)
-		return s.persistAndAnnounceState(globalCtx)
+		var publishedSweep bool
+		for {
+			select {
+			case notification := <-s.blockEpochChan:
+				s.height = notification.(int32)
+				if s.height >= s.LoopInContract.CltvExpiry {
+					err := s.publishIncorrectAmountSweep(
+						globalCtx, htlcOutpoint,
+						htlcValue,
+					)
+					if err != nil {
+						return err
+					}
+
+					publishedSweep = true
+					break
```

Comment:
Would it make sense to republish each block?

Reply:
I think it may be cleaner if we unify this sweeping with the normal timeout sweep. We'd then republish on each block too.

Reply:
Sounds good, will take a look.